### PR TITLE
feat(provider): added simple-texting SMS provider

### PR DESCRIPTION
**Pull request by parveshsaini**
### What change does this PR introduce?

- Introduces SimpleTexting SMS provider
- Doc Reference: https://simpletexting.com/api/docs/v2/

### Why was this change needed?

- Closes: https://github.com/novuhq/novu/issues/4342

### Other information (Screenshots)

image

image

### CODE REVIEW

**Review by Cliftonz — APPROVED** [review]

**Comment by Cliftonz** [comment]
@parveshsaini looks like there are some conflicts to fix.

**Comment by parveshsaini** [comment]
> @parveshsaini looks like there are some conflicts to fix.

done 🚀

**Review by BiswaViraj — CHANGES_REQUESTED** [review]
Looks good,
Can you please fix the failing tests and also add the provider to this list [here](https://github.com/novuhq/novu/blob/524f18bde28296d2e706ca05762d3bb1b28f6b70/apps/web/src/pages/integrations/components/multi-provider/sort-providers.ts#L71C35-L71C35)

File: providers/simpletexting/package.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    "watch:build": "tsc -p tsconfig.json -w",
+    "watch:test": "jest src --watch",
+    "reset-hard": "git clean -dfx && git reset --hard && yarn",
+    "prepare-release": "run-s reset-hard test"
+  },
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@novu/stateless": "0.16.3",
```
Comment (BiswaViraj) [inline providers/simpletexting/package.json:32]:
```suggestion
    "@novu/stateless": "0.21.0",
```

**Review by BiswaViraj — CHANGES_REQUESTED** [review]

**Comment by parveshsaini** [comment]
hey @BiswaViraj 
Can you please guide me on how to fix the failing tests

**Comment by BiswaViraj** [comment]
@parveshsaini, you can ignore the other failing tests as those are something from our side,
For now please fix the spell check test by adding the word to the .cspell.json file

**Comment by parveshsaini** [comment]
hi @BiswaViraj 
I've updated .cspell.json

**Review by Cliftonz — APPROVED** [review]

**Comment by Cliftonz** [comment]
@parveshsaini  looks like we need to resolve some conflicts, this may happen a few times so just be prepared.

**Comment by parveshsaini** [comment]
> @parveshsaini looks like we need to resolve some conflicts, this may happen a few times so just be prepare

🫡

**Comment by parveshsaini** [comment]
conflicts resolved 🚀
@BiswaViraj @Cliftonz

**Comment by BiswaViraj** [comment]
@parveshsaini Cool will merge it once the test CI finishes

**Review by BiswaViraj — APPROVED** [review]

**Pull request merged**